ios/FluentUI/Core/Colors2.swift

```diff
@@ -12,166 +12,110 @@ import UIKit
-        case .brandBackground1Pressed:
-            return .brandBackground1Pressed
-        case .brandBackground1Selected:
-            return .brandBackground1Selected
-        case .brandBackground2:
-            return .brandBackground2
-        case .brandBackground2Pressed:
-            return .brandBackground2Pressed
-        case .brandBackground2Selected:
-            return .brandBackground2Selected
-        case .brandBackground3:
-            return .brandBackground3
-        case .brandBackgroundTint:
-            return .brandBackgroundTint
-        case .brandBackgroundDisabled:
-            return .brandBackgroundDisabled
-        case .brandForeground1:
-            return .brandForeground1
-        case .brandForeground1Pressed:
-            return .brandForeground1Pressed
-        case .brandForeground1Selected:
-            return .brandForeground1Selected
-        case .brandForegroundTint:
-            return .brandForegroundTint
-        case .brandForegroundDisabled1:
-            return .brandForegroundDisabled1
-        case .brandForegroundDisabled2:
-            return .brandForegroundDisabled2
-        case .brandStroke1:
-            return .brandStroke1
-        case .brandStroke1Pressed:
-            return .brandStroke1Pressed
-        case .brandStroke1Selected:
-            return .brandStroke1Selected
-        }
-    }
-}
-
 public extension FluentTheme {
-    /// Associates a `ColorProvider2` with a given `FluentTheme` instance.
+    /// Associates a `ColorProvider2` with a given `FluentThemeable` instance.
     ///
     /// - Parameters:
     ///   - provider: The `ColorProvider2` whose colors should be used for controls in this theme.
-    @objc func setProvider(provider: ColorProviding2) {
+    @objc(setProvider:forThemeable:)
+    static func setProvider(_ provider: ColorProviding2, for themeable: FluentThemeable) {
         // Create an updated fluent theme as well
-        let brandColors = brandColorOverrides(provider: provider, for: self)
+        let brandColors = brandColorOverrides(provider: provider, for: themeable)
+        let fluentTheme = FluentTheme()
         brandColors.forEach { token, value in
-            self.aliasTokens.colors[token] = value
+            fluentTheme.aliasTokens.colors[token] = value
         }
+        themeable.fluentTheme = fluentTheme
     }
 
-    /// Removes any associated `ColorProvider2` from the given `FluentTheme` instance.
-    @objc func removeProvider() {
-        for brandColor in BrandColorsForOverriding.allCases {
-            self.aliasTokens.colors.removeOverride(brandColor.equivalentAliasToken)
-        }
+    /// Removes any associated `ColorProvider` from the given `FluentThemeable` instance.
+    ///
+    /// - Parameters:
+    ///   - window: The `FluentThemeable` that should have its `ColorProvider2` removed.
+    @objc(removeProviderForThemeable:)
+    static func removeProvider(for themeable: FluentThemeable) {
```

Comment:
Should we consider renaming this function to something like `resetThemeable`? `removeProvider` seems a bit misleading since the body of the function doesn’t directly interact with a provider.

Reply:
I was wondering that too - I think I'd actually like to update this entirely:
- Make the methods be explicit categories on UIView, instead of static methods.
- Rename the methods to be very clear:
```swift
@objc(setColorProvider:)
func setColorProvider(_ provider: ColorProviding2)

@objc(resetFluentTheme)
func resetFluentTheme()
```

Sound good?

Reply:
Sounds great, thanks!